### DISCUSSION TITLE
pull backup in a local machine

### DISCUSSION THREAD

**Discussion by migs35323** [longhorn/longhorn#5803] [Q&A]
i have a s3 storage that i use for longhorn with plenty of backups is there a way to get the backups with docker without deploying a cluster in my machine?

i am looking for something similair to this tutorial:
https://longhorn.io/docs/1.4.1/advanced-resources/data-recovery/export-from-replica/
that basically uses docker to mount the volume

it would be ideal if we can use the longhorn interface to pull the backups that we need and select them

**Comment by innobead** [comment]
I don't think we can do that with any interface, so basically you would like to pull the content of a backup w/o Longhorn, right? What's the purpose of that? I would like to know more. Thanks.

**Comment by migs35323** [comment]
@innobead

probbaly it would be possible to launch longhorn in docker/podman and use it for theese special cases ?

i had been using longhorn for a year and there was several situations where i could use this function
there were situations where i had the whole cluster down, or no access to it
and i needed to get access to a volume in the backups to access to the data while we would restart the cluster

in the event that corrupted data gets into the backup, i could pull the volume and make some checks to make sure everything is correct with the volume, so i could sort and discard bad backups and make sure everything is going ok with them

more quickly retrieve and easier access to data basically...
as mentioned before i once had a cluster down situation and instead of launching a docker container to get the data, i had to get a vm running, install k3s (or other flavour of k8s) then install and setup longhorn just to be able to get a single volume

i really could use something similair like the export replica tutorial..
i know we now have the disaster recovery functions but still i would like to basically use docker or other container solution to access to the stuff

i once tried to access the s3 directly to get some files quickly 
but sadly longhorn creates the backups in a format that i didn't know how to use/mount it, and i ended up having to setup a kubernetes local cluster for that one job of retrieving a volume

**Comment by weizhe0422** [comment]
@migs35323 If I understand correctly, maybe you can refer to this [document](https://longhorn.io/docs/1.4.1/advanced-resources/data-recovery/recover-without-system/) to turn the backup into an image file, raw or qcow2, and mount the image file to access the files inside. Hope it can give you some help.
